tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -160,6 +161,26 @@ bool HasHLSLMatOrientation(clang::QualType type, bool *pIsRowMajor) {
   return false;
 }
 
+bool IsHLSLMatRowMajor(clang::QualType type, bool defaultValue) {
+  bool result = defaultValue;
+  HasHLSLMatOrientation(type, &result);
+  return result;
+}
+
+bool IsHLSLUnsigned(clang::QualType type) {
+  if (type->getAs<clang::BuiltinType>() == nullptr) {
+    type = type.getCanonicalType().getNonReferenceType();
+
+    if (IsHLSLVecMatType(type))
+      type = GetElementTypeOrType(type);
+
+    if (type->isExtVectorType())
+      type = type->getAs<clang::ExtVectorType>()->getElementType();
```

Comment:
I don't think we should ever see clang::ExtVectorType in HLSL, right?  Though I can understand including this for completeness I guess.

Reply:
Ok, I say leave it for now.  I'm just not sure I fully understand what scenarios hit certain cases in some of this code.